kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/encryption/VaultTemplateInvocationContext.java

```diff
@@ -0,0 +1,148 @@
+                                            return CompletableFuture.failedStage(e);
+                                        }
+                                    }
+
+                                    @Override
+                                    public CompletionStage<Void> rotateKek(String alias) {
+                                        Objects.requireNonNull(alias);
+                                        try {
+                                            rotate(alias);
+                                            return CompletableFuture.completedStage(null);
+                                        }
+                                        catch (Exception e) {
+                                            // differentiate exceptions
+                                            return CompletableFuture.failedStage(e);
+                                        }
+                                    }
+
+                                    private VaultResponse.ReadKeyData create(String keyId) {
+                                        return runVaultCommand(new TypeReference<>() {
+                                        }, "vault", "write", "-f", "transit/keys/%s".formatted(keyId));
+                                    }
+
+                                    private VaultResponse.ReadKeyData rotate(String keyId) {
+                                        return runVaultCommand(new TypeReference<>() {
+                                        }, "vault", "write", "-f", "transit/keys/%s/rotate".formatted(keyId));
+                                    }
+
+                                    private <D> D runVaultCommand(TypeReference<VaultResponse<D>> valueTypeRef, String... args) {
+                                        try {
+                                            var execResult = vaultContainer.execInContainer(args);
+                                            int exitCode = execResult.getExitCode();
+                                            assertThat(exitCode).isZero();
+                                            var response = new ObjectMapper().readValue(execResult.getStdout(), valueTypeRef);
+                                            return response.data();
+                                        }
+                                        catch (IOException e) {
+                                            throw new UncheckedIOException("Failed to run vault command: %s".formatted(Arrays.stream(args).toList()), e);
+                                        }
+                                        catch (InterruptedException e) {
+                                            Thread.currentThread().interrupt();
+                                            throw new RuntimeException(e);
+                                        }
+                                    }
+
+                                };
+                            }
+                        };
+                    }
+                },
```

Comment:
as in the other comment, I think we should extract the TestContext resolver to make the extension list easier to read